src/core/quantum_gate.jl

```diff
@@ -2179,10 +2205,13 @@ Base.:*(M::Gate, x::Ket) = get_transformed_state(x, M)
 
 function get_transformed_state(state::Ket, gate::Gate)
     transformed_state = deepcopy(state)
-    apply_gate!(transformed_state, gate)
+    apply_instruction!(transformed_state, gate)
     return transformed_state
 end
 
+Base.:*(M::InstructionType, x::Ket) where {InstructionType<:AbstractInstruction} =
```

Comment:
since we have `Base.:*(M::Gate, x::Ket) = get_transformed_state(x, M)`, and Gate is now a AbstractInstruction, it made sense to have an explicit fallback for any other `AbstractInstruction`. If we take this out we would simply get a MethodError for `Base.:*(::Readout,::Ket)`. Do you think that is better?